modules/stochastic_tools/include/utils/GaussianProcessHandler.h

```diff
@@ -50,13 +50,16 @@ class GaussianProcessHandler
    *                      covariance matrix.
    * @param opt_type The optimization algorithm for the hyperparameters.
    * @param tao_options Additional options if TAO is used for parameter optimization.
-   * @param show_tao Switch to show details of TAO optimization.
+   * @param show_tao_adam Switch to show details of TAO optimization.
    */
   void setupCovarianceMatrix(const RealEigenMatrix & training_params,
                              const RealEigenMatrix & training_data,
                              MooseEnum opt_type,
                              std::string tao_options = "",
-                             bool show_tao = false);
+                             bool show_tao_adam = false,
```

Comment:
I still think we should group these parameters into a struct, something like this:

```
struct GPOptimizerOptions
{
   MooseEnum opt_type(...);
  std::string tao_options;
  unsigned int iter_adam;
  ...
}
```
and then you could just use it as follows:

```
GPOptimizerOptions opts;
opts.method = "adam";
opts.learning_rate=1e-4;
...
setupCovarianceMatrix(training_params, training_data, opts);
```

So your function would be something like this:

```
void setupCovarianceMatrix(const RealEigenMatrix & training_params,
                                               const RealEigenMatrix & training_data,
                                               const GPOptimizerOptions & opts,
```

Reply:
The opt-type could not be declared in a struct and passed to the Handler function. It has to do with it being ENUM:

moose/modules/stochastic_tools/src/surrogates/GaussianProcessTrainer.C:181:63: error: call to implicitly-deleted default constructor of 'StochasticTools::GaussianProcessHandler::GPOptimizerOptions'
  StochasticTools::GaussianProcessHandler::GPOptimizerOptions opts;
                                                              ^
moose/modules/stochastic_tools/build/header_symlinks/GaussianProcessHandler.h:57:15: note: default constructor of 'GPOptimizerOptions' is implicitly deleted because field 'opt_type' has an inaccessible default constructor
    MooseEnum opt_type;

Reply:
@zachmprince , do you think this is something we should add or let's just roll without it? If I fix the documentation, I can add this one too.